asf_welsh_energy_consultation/pipeline/process_data.py

```diff
@@ -0,0 +1,256 @@
+# File: asf_welsh_energy_consultation/pipeline/process_data.py
+"""
+Functions to process and augment data.
+"""
+
+import pandas as pd
+
+from asf_welsh_energy_consultation.getters.get_data import *
+
+
+# PROCESSING MCS
+
+
+def get_enhanced_mcs():
+    """Get dataset of domestic MCS installations with attached off-gas, country and rurality fields.
+
+    Returns:
+        pd.DataFrame: Dataset as described above.
+    """
+    mcs = get_mcs_domestic()
+    og = get_offgas()
+    countries = get_countries()
+    rural = get_rurality()
+
+    # join with off-gas data
+    mcs = mcs.merge(og, on="postcode", how="left")
+    mcs["off_gas"] = mcs["off_gas"].fillna("On gas").replace({True: "Off gas"})
+
+    # join with regions in order to filter to Wales
+    mcs = mcs.merge(countries, on="postcode", how="left")
+    mcs = mcs.loc[mcs["country"] == "Wales"].reset_index(drop=True)
+    # 1203 records with no match - 273 are Northern Ireland which leaves 918
+    # Some will be new postcodes (new build developments)
+    # and some may be expired postcodes
+    # In future, implement new solution that uses outward codes
+
+    # join with rurality data
+    mcs = mcs.merge(rural, on="postcode", how="left")
+    # only 13 postcodes lost in this merge
+
+    # add custom rurality column
+    mcs["rurality_7"] = mcs["rurality_10_label"].replace(
+        {
+            "Urban city and town": "Urban",
+            "Urban major conurbation": "Urban",
+            "Urban city and town in a sparse setting": "Urban",
+            "Urban minor conurbation": "Urban",
+        }
+    )
+
+    return mcs
+
+
+# load enhanced MCS as part of this script, so only needs to be done once
+mcs = get_enhanced_mcs()
```

Comment:
It's used in `get_mcs_retrofits` as well, so I defined it separately to avoid loading it twice. I'll action your suggestion of calling it `enhanced_mcs`